File: 240. Search a 2D Matrix II/searchMatrix.py
class Solution(object):
    def searchMatrix(self, matrix, target):
        """
        :type matrix: List[List[int]]
        :type target: int
        :rtype: bool
        """
        if len(matrix) == 0:
            return False
            
        row = 0
        col = len(matrix[0]) - 1
        
        while row < len(matrix) and col >= 0:
            val = matrix[row][col]
            if target == val:
                return True
            if target > val:
                row += 1
            if target < val:
                col -= 1

        return False

File: 240. Search a 2D Matrix II/searchMatrix.java
class Solution {
    //Start from top-right corner, and only shift down or left if necessary.
    //If the current value in current grid is same as target, return true;
    //if the current value in current grid is bigger than target, move left;
    //if the current value in current grid is smaller than target, move down.
    public boolean searchMatrix(int[][] matrix, int target) {
        int row = matrix.length;
        if(row == 0)    return false;
        int col = matrix[0].length;
        if(col == 0)    return false;
        
        col -= 1;
        int currentRow = 0;
        while(col >= 0 && currentRow < row){
            if(matrix[currentRow][col] == target)
                return true;
            else if (matrix[currentRow][col] < target)
                currentRow++;
            else
                col--;
        }
        return false;
    }
}